src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/UpdateValidator.java

```diff
@@ -15,27 +19,45 @@
 @Component
 public class UpdateValidator {
 
-    public Errors validate(UpdateDto updateDto, Errors errors, String loggingContext) {
+    @Autowired
+    private PublicDataRetrievalService publicDataRetrievalService;
+
+    public Errors validate(String entityNumber, UpdateDto updateDto, Errors errors, String loggingContext, String passThroughTokenHeader) throws ServiceException {
         if (updateDto.getFilingDate() != null) {
-            validateFilingDate(updateDto.getFilingDate(), errors, loggingContext);
+            validateFilingDate(entityNumber, updateDto.getFilingDate(), errors, loggingContext, passThroughTokenHeader);
         }
         return errors;
     }
 
-    public Errors validateFull(UpdateDto updateDto, Errors errors, String loggingContext) {
+    public Errors validateFull(String entityNumber, UpdateDto updateDto, Errors errors, String loggingContext, String passThroughTokenHeader) throws ServiceException {
         if (updateDto == null || updateDto.getFilingDate() == null) {
             String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.UPDATE_FIELD, UpdateDto.FILING_DATE);
             String errorMessage = ValidationMessages.NOT_NULL_ERROR_MESSAGE.replace("%s", qualifiedFieldName);
             setErrorMsgToLocation(errors, qualifiedFieldName, errorMessage);
             ApiLogger.infoContext(loggingContext, errorMessage);
         } else {
-            validateFilingDate(updateDto.getFilingDate(), errors, loggingContext);
+            validateFilingDate(entityNumber, updateDto.getFilingDate(), errors, loggingContext, passThroughTokenHeader);
         }
         return errors;
     }
 
-    private boolean validateFilingDate(LocalDate filingDate, Errors errors, String loggingContext) {
+    private void validateFilingDate(String entityNumber, LocalDate filingDate, Errors errors, String loggingContext, String passThroughTokenHeader) throws ServiceException {
         String qualifiedFieldName = getQualifiedFieldName(OverseasEntitySubmissionDto.UPDATE_FIELD, UpdateDto.FILING_DATE);
-        return DateValidators.isDateInPast(filingDate, qualifiedFieldName, errors, loggingContext);
+        DateValidators.isDateInPast(filingDate, qualifiedFieldName, errors, loggingContext);
+
+        ApiLogger.debugContext(loggingContext, "Check filing date of " + filingDate + " for ROE company "
+                + entityNumber + " against the next Made Up To date");
+
+        var companyProfileApi = publicDataRetrievalService.getCompanyProfile(entityNumber, passThroughTokenHeader);
+        LocalDate nextMadeUpToDate = companyProfileApi.getConfirmationStatement().getNextMadeUpTo();
```

Comment:
companyProfileApi.getConfirmationStatement() might be null here for some OEs? Would you add a ull check or ust let the exception be thrown and caught later?

Reply:
Good call. Will have a think about this.

Reply:
Decided to check for null and throw a `ServiceException` if no details returned.